keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -116,6 +117,43 @@ class ClusterHandler(
       .getActiveServerGroups(resource)
       .byRegion()
 
+  override suspend fun willTakeAction(
+    resource: Resource<ClusterSpec>,
+    resourceDiff: ResourceDiff<Map<String, ServerGroup>>
+  ): ActionResponse {
+    // we can't take any action if there is more than one active server group
+    //  AND the current active server group is unhealthy
+    val potentialInactionableRegions = mutableListOf<String>()
+    val inactionableRegions = mutableListOf<String>()
+    resourceDiff.toIndividualDiffs().forEach { diff ->
+      if (diff.hasChanges() && diff.isEnabledOnly()) {
+        potentialInactionableRegions.add(diff.desired.location.region)
+      }
+    }
+    if (potentialInactionableRegions.isNotEmpty()) {
+      val activeServerGroups = cloudDriverService.getActiveServerGroups(resource)
+      activeServerGroups.forEach { serverGroup ->
+        val healthy = serverGroup.instanceCounts?.isHealthy(
+          resource.spec.deployWith.health,
+          resource.spec.resolveCapacity(serverGroup.location.region)
+        ) == true
+        if (!healthy && potentialInactionableRegions.contains(serverGroup.location.region)) {
+          inactionableRegions.add(serverGroup.location.region)
+        }
+      }
+      if (inactionableRegions.isNotEmpty()) {
```

Comment:
Here's the message we send for the case where we can't do anything.

Note: these methods are almost identical for the cluster handlers. It's on my todo list to create a nice base class for cluster handlers. I dream about it. But it hasn't happened yet.